docs/sections/haxe.md

```diff
@@ -0,0 +1,24 @@
+# Haxe (`haxe`)
+
+!!! important "This section is rendered asynchronously by default"
+
+!!! info
+[**Haxe**](https://haxe.org) is an open source high-level strictly-typed programming language with a fast optimizing cross-compiler.
+
+The `haxe` section displays the Haxe version.
+
+This section is displayed only when:
+
+- Upsearch finds `[P/p]roject.xml`, `application.xml`, `haxelib.json`, `hxformat.json` or `.haxerc` file
+- Current directory contains `*.hxml` or `*.hx`, files.
+
+## Options
+
+| Variable                |              Default               | Meaning                             |
+| :---------------------- | :--------------------------------: | ----------------------------------- |
+| `SPACESHIP_HAXE_SHOW`   |               `true`               | Show section                        |
+| `SPACESHIP_HAXE_ASYNC`  |               `true`               | Render section asynchronously       |
+| `SPACESHIP_HAXE_PREFIX` |               `with`               | Section's prefix                    |
+| `SPACESHIP_HAXE_SUFFIX` | `$SPACESHIP_PROMPT_DEFAULT_SUFFIX` | Section's suffix                    |
+| `SPACESHIP_HAXE_SYMBOL` |                `⌘`                 | Symbol displayed before the section |
+| `SPACESHIP_HAXE_COLOR`  |           `166` (orange)           | Section's color                     |
```

Comment:
Add docs for `SPACESHIP_HAXE_VERBOSE` option. Preferably, with an example displaying the option. Refer to other sections, like https://spaceship-prompt.sh/sections/rust/?h=#Displaying-verbose-version

Reply:
Yes, but `haxe --version` command doesn't yield any verbose result other than just the version number as `4.3.1`.

Reply:
Should we add it in the docs, even if the output remains the same?

Reply:
Then what does this line do?

```
[[ $SPACESHIP_HAXE_VERBOSE == false ]] && haxe_version=${haxe_version%-*}
```

https://github.com/spaceship-prompt/spaceship-prompt/pull/1367/files#diff-335c8b7ab5d29c74c0594739294a74827528d3dd3dc7e78f7119f58ecec40b39R40

Should we remove it them?

Reply:
Done, @denysdovhan, Awaiting Review.